[docs] add information about submit credentials to `security` docs (#30628)
expo/expo: docs/pages/app-signing/security.mdx

Expo tools never ask you to provide your Google account credentials.

## Android submit credentials

### Google Service Account Key

> Google Service Account Key is the authentication method required to be used to submit an Android app to the Google Play Store using the EAS Submit service. This key is permanently stored on the Expo servers (unless deleted by a developer) while being encrypted using [KMS](https://cloud.google.com/security/products/security-key-management) when at rest.

Reviewer: ```suggestion
Google Service Account Key is the authentication method used to submit an Android app to the Google Play Store with EAS Submit. This key is stored on Expo servers and encrypted using [KMS](https://cloud.google.com/security/products/security-key-management) when at rest. The key will remain on Expo servers to be re-used for subsequent submissions, and it can be removed at any time by a user with the requisite permissions.
```

#### Consequences if compromised

> If a malicious actor somehow gains access to your Google Service Account Key, they would be able to perform actions in the Google Play Console on your behalf. The actions they could perform would be limited to the permissions granted to the service account key. They would also be able to submit an existing app signed by your build credentials to the Google Play Store.

Reviewer: ```suggestion
If a malicious actor somehow gains access to your Google Service Account Key, they would be able to perform actions in the Google Play Console on your behalf. The actions they could perform would be limited to the permissions granted to the service account key.

If the attacker has additionally gained access to your upload keystore, they would be able to submit a new version of an existing app. The actor would not be able to submit a new app to the Google Play Store in your name, as the first Google Play submission needs to be done through the web console.
```

> If an attacker doesn't have access to your build credentials, they won't be able to do much when it comes to submitting an existing app to the Google Play Store. They would only be able to submit the app signed with them. The actor also wouldn't be able to submit a new app to the Google Play Store in your name, as the first Google Play submission needs to be done manually.

Reviewer: ```suggestion
```

Expo tools never ask you to provide your Google account credentials.

## Android submit credentials

### Google Service Account Key

Google Service Account Key is the authentication method used to submit an Android app to the Google Play Store with EAS Submit. This key is stored on Expo servers and encrypted using [KMS](https://cloud.google.com/security/products/security-key-management) when at rest. The key will remain on Expo servers to be re-used for subsequent submissions, and it can be removed at any time by a user with the requisite permissions.

#### Consequences if compromised

If a malicious actor somehow gains access to your Google Service Account Key, they would be able to perform actions in the Google Play Console on your behalf. The actions they could perform would be limited to the permissions granted to the service account key.

If the attacker has additionally gained access to your upload keystore, they would be able to submit a new version of an existing app. The actor would not be able to submit a new app to the Google Play Store in your name, as the first Google Play submission needs to be done through the web console.

> 

Author: ```suggestion
```

Expo tools never ask you to provide your Google account credentials.

## Android submit credentials

### Google Service Account Key

Google Service Account Key is the authentication method required to be used to submit an Android app to the Google Play Store using the EAS Submit service.

> The Google Service Account Key is permanently stored on the Expo servers (unless deleted by a developer) while being encrypted using [KMS](https://cloud.google.com/security/products/security-key-management) when at rest.

Author: >Also, what does "when at rest" mean here? Is it like when it's not in use?

Yeah, that's what I meant

#### Consequences if compromised

> If a malicious actor were to somehow gain access to the Google Service Account Key, they would be able to perform actions in the Google Play Console on your behalf. The actions they could perform would be limited to the permissions granted to the service account key.
> They would also be able to submit an existing app signed by your build credentials to the Google Play Store.
> When it comes to submitting an existing app to the Google Play Store, if an attacker doesn't have access to your build credentials, they won't be able to do much, as they would only be able to submit the app signed with them.
> The actor also wouldn't be able to submit a new app to the Google Play Store in your name, as the first Google Play submission needs to be done manually.

Reviewer 2: ```suggestion
If a malicious actor somehow gains access to your Google Service Account Key, they would be able to perform actions in the Google Play Console on your behalf. The actions they could perform would be limited to the permissions granted to the service account key. They would also be able to submit an existing app signed by your build credentials to the Google Play Store.

If an attacker doesn't have access to your build credentials, they won't be able to do much when it comes to submitting an existing app to the Google Play Store. They would only be able to submit the app signed with them. The actor also wouldn't be able to submit a new app to the Google Play Store in your name, as the first Google Play submission needs to be done manually.
```

I think there are two different parts to explain in this section. One is someone gaining access to Google Service Account Key, and the second is submitting them to Google Play Store and what happens if the attacker doesn't have access to the build credentials? Let's divide them into different paragraphs to improve the readability.

None. They are available through the Apple Developer console.

## iOS submit credentials

### Apple App Store Connect (ASC) API key

> Apple App Store Connect (ASC) API key is one of the authentication methods that can be used to submit an iOS app to Apple's App Store using the EAS Submit service. This key is permanently stored on the Expo servers (unless deleted by a developer) while being encrypted using [KMS](https://cloud.google.com/security/products/security-key-management) when at rest.

Reviewer: ```suggestion
Apple App Store Connect (ASC) API key is one of the authentication methods that can be used to submit an iOS app to Apple's App Store using the EAS Submit service. This key is stored on the Expo servers and encrypted using [KMS](https://cloud.google.com/security/products/security-key-management) when at rest. The key will remain on Expo servers to be re-used for subsequent submissions, and it can be removed at any time by a user with the requisite permissions.
```

The ASC API key is the default and **recommended** authentication method for submitting your apps to the App Store using EAS Submit.

#### Consequences if compromised

> If a malicious actor somehow gains access to the ASC API key, they would be able to perform actions in the App Store Connect on your behalf. The actions they could perform would be limited to the permissions granted to the API key. They would also be able to submit an app signed by your build credentials to the App Store.

Reviewer: ```suggestion
If a malicious actor somehow gains access to the ASC API key, they would be able to perform actions in the App Store Connect on your behalf. The actions they could perform would be limited to the permissions granted to the API key.
```

> If an attacker doesn't have access to your build credentials, they won't be able to do much when submitting an app to the App Store. They can only submit the app signed with those build credentials, and they can't submit any arbitrary app to the App Store in your name.

Reviewer: ```suggestion
If the attacker has additionally gained access to your build credentials, they would be able to submit a new version of an existing app. They would only be able to submit the app signed with those build credentials, and they can't submit any arbitrary app to the App Store in your name.
```

#### Consequences if lost

None. If you lose the ASC API key, you can revoke it using the App Store Connect portal and create a new one.

### Apple app-specific password

> Apple app-specific password is another authentication method that can be used to submit an iOS app to Apple's App Store using the EAS Submit. Unlike other credentials, the app-specific password is not permanently stored in the Expo servers.

Reviewer: ```suggestion
Apple app-specific password is another authentication method that can be used to submit an iOS app to Apple's App Store using the EAS Submit. Unlike other credentials, the app-specific password is not stored in the Expo servers between submissions, it must be provided each time it is to be used.
```

> The password is encrypted using [KMS](https://cloud.google.com/security/products/security-key-management) and stored only for the period required to submit the app to the App Store plus 24 hours for retries. Once this period is over, the password is deleted from the Expo servers.

Reviewer: ```suggestion
The password is encrypted using [KMS](https://cloud.google.com/security/products/security-key-management) and stored only for the period required to submit the app to the App Store plus 24 hours, to allow for retries during that time. Once this period is over, the password is deleted from the Expo servers.
```

This authentication method is **not recommended**. We recommend using the Apple Store Connect (ASC) API key for submitting your apps to the App Store instead. Expo won't use the Apple app-specific password in any way other than to submit your app to the App Store.

#### Consequences if compromised

> If a malicious actor somehow gains access to the app-specific password, they would be able to access information like mail, contacts, and calendars that you store in iCloud (check [Apple's documentation](https://support.apple.com/en-us/102654) for more details). They would also be able to submit an app signed by your build credentials to the App Store.

Reviewer: ```suggestion
If a malicious actor somehow gains access to the app-specific password, they would be able to access information like mail, contacts, and calendars that you store in iCloud (check [Apple's documentation](https://support.apple.com/en-us/102654) for more details).
```

> If an attacker doesn't have access to your build credentials, they won't be able to do much when submitting an app to the App Store. They can only submit the app signed with those build credentials, and they can't submit any arbitrary app to the App Store in your name.

Reviewer: ```suggestion
If the attacker has additionally gained access to your build credentials, they would be able to submit a new version of an existing app. They would only be able to submit the app signed with those build credentials, and they can't submit any arbitrary app to the App Store in your name.
```

None. They are available through the Apple Developer console.

## iOS submit credentials

### Apple App Store Connect (ASC) API key

> Apple App Store Connect (ASC) API key authentication is one of the authentication methods that can be used to submit an iOS app to Apple's App Store using the EAS Submit service.
> The ASC API key is permanently stored on the Expo servers (unless deleted by a developer) while being encrypted using [KMS](https://cloud.google.com/security/products/security-key-management) when at rest.
> The ASC API key is the default and **recommended** authentication method for submitting your apps to the App Store using EAS Submit.

Reviewer 2: ```suggestion
Apple App Store Connect (ASC) API key is one of the authentication methods that can be used to submit an iOS app to Apple's App Store using the EAS Submit service. This key is permanently stored on the Expo servers (unless deleted by a developer) while being encrypted using [KMS](https://cloud.google.com/security/products/security-key-management) when at rest.

The ASC API key is the default and **recommended** authentication method for submitting your apps to the App Store using EAS Submit.
```

#### Consequences if compromised

> If a malicious actor were to somehow gain access to the ASC API key, they would be able to perform actions in the App Store Connect on your behalf. The actions they could perform would be limited to the permissions granted to the API key.
> They would also be able to submit an app signed by your build credentials to the App Store.
> When it comes to submitting an app to the App Store, if an attacker doesn't have access to your build credentials, they won't be able to do much, as they would only be able to submit the app signed with them.
> It means that they wouldn't be able to submit any arbitrary app to the App Store in your name.

Reviewer 2: ```suggestion
If a malicious actor were to somehow gain access to the ASC API key, they would be able to perform actions in the App Store Connect on your behalf. The actions they could perform would be limited to the permissions granted to the API key. They would also be able to submit an app signed by your build credentials to the App Store.

If an attacker doesn't have access to your build credentials, they won't be able to do much when submitting an app to the App Store. They can only submit the app signed with those build credentials, and they can't submit any arbitrary app to the App Store in your name.
```

#### Consequences if lost

None. If you lose the ASC API key, you can revoke it using the App Store Connect portal and create a new one.

### Apple app-specific password

> Apple app-specific password authentication is one of the authentication methods that can be used to submit an iOS app to Apple's App Store using the EAS Submit service.
> Unlike other credentials, the app-specific password is not permanently stored in the Expo servers. The password is encrypted using [KMS](https://cloud.google.com/security/products/security-key-management) and stored only for the period required to submit the app to the App Store + 24 hours for retries.
> Once this period is over, the password is deleted from the Expo servers. This authentication method is **not recommended**. We recommend using the Apple Store Connect (ASC) API key for submitting your apps to the App Store instead.
> Expo won't use the Apple app-specific password in any other way than for submitting your app to the App Store.

Reviewer 2: ```suggestion
Apple app-specific password is another authentication method that can be used to submit an iOS app to Apple's App Store using the EAS Submit. Unlike other credentials, the app-specific password is not permanently stored in the Expo servers. 

The password is encrypted using [KMS](https://cloud.google.com/security/products/security-key-management) and stored only for the period required to submit the app to the App Store plus 24 hours for retries. Once this period is over, the password is deleted from the Expo servers. 

This authentication method is **not recommended**. We recommend using the Apple Store Connect (ASC) API key for submitting your apps to the App Store instead. Expo won't use the Apple app-specific password in any way other than to submit your app to the App Store.
```

Let' improve structure of this section for better readability.

#### Consequences if compromised

> If a malicious actor were to somehow gain access to the app-specific password, they would be able to access information like mail, contacts, and calendars that you store in iCloud.
> They would also be able to submit an app signed by your build credentials to the App Store.
> When it comes to submitting an app to the App Store, if an attacker doesn't have access to your build credentials, they won't be able to do much, as they would only be able to submit the app signed with them.
> It means that they wouldn't be able to submit any arbitrary app to the App Store in your name.

Reviewer 2: ```suggestion
If a malicious actor were to somehow gain access to the app-specific password, they would be able to access information like mail, contacts, and calendars that you store in iCloud. They would also be able to submit an app signed by your build credentials to the App Store.

If an attacker doesn't have access to your build credentials, they won't be able to do much when submitting an app to the App Store. They can only submit the app signed with those build credentials, and they can't submit any arbitrary app to the App Store in your name.
```
